xbmc/addons/VFSEntry.cpp

```diff
@@ -70,8 +70,18 @@ VFSEntryPtr CVFSAddonCache::GetAddonInstance(const std::string& strId, TYPE type
 
 void CVFSAddonCache::OnEvent(const AddonEvent& event)
 {
-  if (typeid(event) == typeid(AddonEvents::InstalledChanged))
+  if (typeid(event) == typeid(AddonEvents::Initilized) ||
+      typeid(event) == typeid(AddonEvents::UnInstalled))
```

Comment:
The `AddonEvents::UnInstalled` also to the check below.

Reply:
As I said `UnInstalled` can't be used in conjunction with `HasType` because the addon is gone